source/service.go

```diff
@@ -92,7 +97,19 @@ func (s *ServiceSource) buildRecord(obj interface{}, action string) (resource.Re
 		return advertiseObj, nil
 	}
 
-	advertiseObj.Name = service.Name
+	if hostnames, ok := service.Annotations["external-mdns.blakecovarrubias.com/hostnames"]; ok {
+		names := strings.Split(hostnames, ",")
+		for i := range names {
+			names[i] = strings.TrimSpace(names[i])
+		}
+		advertiseObj.Names = names
```

Comment:
Since this doesn't perform any other sanitization on the hostname other than removing spaces, if a user as a service in the default namespace and sets their hostname to `foo.local`, the resultant record will be `foo.local.local`.

At the very least the README should be updated to clarify this behavior and recommend users omit the top-level `.local` domain.

Long term, we should probably add some input validation to ensure that the field only accepts valid DNS hostnames, including [IDNs](https://en.wikipedia.org/wiki/Internationalized_domain_name).